app/post.go

```diff
@@ -250,8 +250,10 @@ func UpdatePost(post *model.Post, safeUpdate bool) (*model.Post, *model.AppError
 			return nil, err
 		}
 
-		if oldPost.IsSystemMessage() {
-			err := model.NewAppError("UpdatePost", "api.post.update_post.system_message.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+		if oldPost.IsSystemMessage() &&
+			!(oldPost.Type == model.POST_JOIN_LEAVE || oldPost.Type == model.POST_JOIN_CHANNEL || oldPost.Type == model.POST_LEAVE_CHANNEL ||
+				oldPost.Type == model.POST_ADD_REMOVE || oldPost.Type == model.POST_ADD_TO_CHANNEL || oldPost.Type == model.POST_REMOVE_FROM_CHANNEL) {
```

Comment:
You've also repeated this twice. Can you just move this expression to a `Post` method? Something like `oldPost.IsJoinLeave()`?